### ISSUE TITLE
check_version 10.1 fails

### ISSUE THREAD

**Issue by maletin** [bucardo/check_postgres#130]
https://github.com/bucardo/check_postgres/blob/a4a1d23a63a54ef0c7e4fe8850b423a7098dd060/check_postgres.pl#L8354
"10.1" is a fullversion, but the above line unsets the variable `$warnfull`.

**Comment by machack666** [comment]
Hi @maletin, I'm not sure under what circumstances this error occurs.  Which local version of Pg should be used and which test is failing?

Thanks,

David

**Comment by maletin** [comment]
```
~$ apt-cache policy postgresql-10
postgresql-10:
  Installiert:           10.1-1.pgdg16.04+1
  Installationskandidat: 10.1-1.pgdg16.04+1
  Versionstabelle:
 *** 10.1-1.pgdg16.04+1 500
        500 http://apt.postgresql.org/pub/repos/apt xenial-pgdg/main amd64 Packages
        100 /var/lib/dpkg/status
$ check_postgres.pl --action=version -w='10'
POSTGRES_VERSION OK: DB "postgres" (port=5434) version 10.1 | time=0.14s
$ check_postgres.pl --action=version -w='10.1.0'
POSTGRES_VERSION WARNING: DB "postgres" (port=5434) version 10.1, but expected 10.1.0 | time=0.08s
$ check_postgres.pl --action=version -w='10.1'
POSTGRES_VERSION WARNING: DB "postgres" (port=5434) version 10.1, but expected 10.1 | time=0.08s
```

**Comment by erlinghaugstad** [comment]
For the following command:
`check_postgres --action new_version_pg`

And with the following output from postgres(select version()):
`PostgreSQL 10.1 on x86_64-pc-linux-gnu, compiled by gcc (Debian 4.9.2-10) 4.9.2, 64-bit`

The result is:
`POSTGRES_NEW_VERSION_PG UNKNOWN: Could not download version information for Postgres`

If i run:
`check_postgres --action=version -w='10.1'`

The result is:
`POSTGRES_VERSION WARNING: DB "postgres" version 10.1, but expected 10.1 | time=0.08s`

The parsing logic fails to identify 10.1 as the version since it is expecting an x.y.z version number. I think it actually uses 4.9.2 since that conforms with what is expected, but that is a non existing postgres version.

If i was half decent in perl i would look into making a pull request, but i guess i have a better chance if someone looks at this :)

**Comment by MichaelDBA** [comment]
I can confirm the same problem against 10.1 on Ubuntu 14.04 LTS:
```
check_postgres --port=5441 --action=version -w='10.1'
POSTGRES_VERSION WARNING: DB "postgres" (port=5441) version 10.1, but expected 10.1 | time=0.01s
```

**Comment by machack666** [comment]
This should be fixed now in 15d07f3a.

**Comment by machack666** [comment]
And fixed a bug in the fix in cc8970e. :-)

**Comment by MichaelDBA** [comment]
Thanks!
